src/Event/EventDispatcherInterface.php

```diff
@@ -33,13 +33,13 @@ interface EventDispatcherInterface
      * Returns a dispatched event.
      *
      * @param string $name Name of the event.
-     * @param array|null $data Any value you wish to be transported with this event to
+     * @param array $data Any value you wish to be transported with this event to
      * it can be read by listeners.
      * @param object|null $subject The object that this event applies to
      * ($this by default).
      * @return \Cake\Event\EventInterface
      */
-    public function dispatchEvent(string $name, ?array $data = null, ?object $subject = null): EventInterface;
+    public function dispatchEvent(string $name, array $data = [], ?object $subject = null): EventInterface;
```

Comment:
Changing interface method signature would be technically a BC break.

Reply:
Yea, but this shows what a better fix is. The default of null was a very odd choice which requires the array cast.